src/java.base/aix/classes/sun/nio/ch/PollsetPoller.java

```diff
@@ -25,39 +26,79 @@
 package sun.nio.ch;
 
 import java.io.IOException;
+import java.time.Instant;
+import sun.nio.ch.Pollset;
 
 /**
  * Poller implementation based on the AIX Pollset library.
  */
 
 class PollsetPoller extends Poller {
 
+    static { Pollset.init(); /* Dynamically loads pollset C functions */ }
+
+    private final int setid;
+    private final long pollBuffer;
+    private static final int POLL_BUFFER_SIZE = 16;
```

Comment:
16 seems very low. On the other platforms the equivalent is sized to read up to 256 or 512 events in one syscall but maybe there is a reason to choose this small number? You might need to run Helidon Nima or some other server using virtual threads to see the impact on this.

The name used on other platforms for the constant is "MAX_EVENTS_TO_POLL" to better reflect the usage.

Reply:
I agree with Alan, when I was in IBM, I re-implemented the same for IBM Z/OS and my initial buffer size was very low. The product team complains that first 512 clients connections  were slow. I increased buffer size to 512( MAX_POLL_EVENTS = 512;) and problem got resolved. Although  Z/OS implementation was based on “select and poll” API which is different than your AIX implementation and my be, buffer size is not applicable to AIX.

Reply:
Thanks for your feedback. I have changed this to reference the existing MAX_POLL_EVENTS constant which is set to 512.